service/dap/server_test.go

```diff
@@ -306,8 +306,139 @@ func TestSetBreakpoint(t *testing.T) {
 			}
 		}
 
+		client.StackTraceRequest(1, 0, 20)
+		stResp := client.ExpectStackTraceResponse(t)
+
+		if stResp.Body.TotalFrames != 6 {
+			t.Errorf("\ngot %#v\nwant TotalFrames=6", stResp.Body.TotalFrames)
+		}
+		if len(stResp.Body.StackFrames) != 6 {
+			t.Errorf("\ngot %#v\nwant len(StackFrames)=6", stResp.Body.StackFrames)
+		} else {
+			expectStackFrame := func(got dap.StackFrame, id int, name string, sourceName string, line int) {
+				t.Helper()
+				if got.Id != id || got.Name != name {
+					t.Errorf("\ngot  %#v\nwant id=%d name=%s", got, id, name)
+				}
+				if (sourceName != "" && got.Source.Name != sourceName) || (line > 0 && got.Line != line) {
+					t.Errorf("\ngot  %#v\nwant source.name=%s line=%d", got, sourceName, line)
+				}
+			}
+			expectStackFrame(stResp.Body.StackFrames[0], 1000, "main.Increment", "increment.go", 8)
+			expectStackFrame(stResp.Body.StackFrames[1], 1001, "main.Increment", "increment.go", 11)
+			expectStackFrame(stResp.Body.StackFrames[2], 1002, "main.Increment", "increment.go", 11)
+			expectStackFrame(stResp.Body.StackFrames[3], 1003, "main.main", "increment.go", 17)
+			expectStackFrame(stResp.Body.StackFrames[4], 1004, "runtime.main", "proc.go", -1)
+			expectStackFrame(stResp.Body.StackFrames[5], 1005, "runtime.goexit", "", -1)
+		}
+
 		// TODO(polina): add other status checking requests
-		// that are not yet supported (stackTrace, scopes, variables)
+		// that are not yet supported (scopes, variables)
+
+		client.ContinueRequest(1)
+		client.ExpectContinueResponse(t)
+		// "Continue" is triggered after the response is sent
+
+		client.ExpectTerminatedEvent(t)
+		client.DisconnectRequest()
+		client.ExpectDisconnectResponse(t)
+	})
+}
+
+// TestStackTraceRequest executes to a breakpoint (similarly to TestSetBreakpoint
+// that includes more thorough checking of that sequence) and tests different
+// good and bad configurations of 'stackTrace' requests.
+func TestStackTraceRequest(t *testing.T) {
```

Comment:
:+1:   Nice comprehensive testing

Reply:
Thanks. Such tests take longer than the first draft of the code, but certainly help iron out the kinks :)